### PR TITLE
Add getMethods to Server

### PR DESCRIPTION
**Pull request by FTI-Herbert**

### CODE REVIEW

File: src/Server.php [hunk 1]
```diff
@@ -90,6 +90,14 @@ public function run(string $raw): ?string
         return $this->single($input);
     }
 
+    /**
+     * @return string[]|null
+     */
+    public function getMethods(): ?array
```
Comment (1ma) [inline src/Server.php:96]: I think the methods attribute cannot be null under any circumstances, better type to `array`

**Review by 1ma — CHANGES_REQUESTED** [review]

**Review by 1ma — APPROVED** [review]

**Comment by 1ma** [comment]
I imagine you'll implement the discovery feature with a procedure and use it along these lines: `$server->set('rpc.discover', RpcDiscovery::class)`. If you want to contribute that procedure feel free to do so.

**Comment by FTI-Herbert** [comment]
> I imagine you'll implement the discovery feature with a procedure and use it along these lines: `$server->set('rpc.discover', RpcDiscovery::class)`. If you want to contribute that procedure feel free to do so.
Your rigth, my implementaition looks similary. I'm not shure if it's possible to contibute it. There are some problerms to solve for this. Right now my implementaiton is to close to laminas. But i think this can be solved with some affort. One of the other problems is, we will not implement the whole OpenRpc stack. Only the parts we use and what is required by the specs. So this will cover only a small part of OpenRpc. So maybe there will be a way to contribute this, but i'm not shure if i can handle this.

**Comment by 1ma** [comment]
No worries.

**Pull request merged**